src/OpenLoco/src/World/Station.cpp

```diff
@@ -1055,6 +1055,40 @@ namespace OpenLoco
         }
     }
 
+    // 0x0048F482
+    void removeTileFromStation(const StationId stationId, const World::Pos3& pos, uint8_t rotation)
+    {
+        auto* station = StationManager::get(stationId);
+        auto findPos = pos;
+        findPos.z |= rotation;
+
+        // Bug mitigation: ensure stationTileSize does not exceed the actual array length
+        station->stationTileSize = std::clamp<uint16_t>(station->stationTileSize, 0, std::size(station->stationTiles));
+
+        // Find tile to remove
+        auto tileIndex = std::find(std::begin(station->stationTiles), std::end(station->stationTiles), findPos);
+        if (tileIndex == std::end(station->stationTiles))
+            return;
+
+        // Remove tile by moving the remaining tiles over the one to remove
+        std::remove(std::begin(station->stationTiles), std::end(station->stationTiles), findPos);
+        station->stationTileSize--;
```

Comment:
This seemed like a short and safe way of erasing the tile registration, if found.